src/Model/PrestaShop.php

```diff
@@ -21,16 +26,26 @@ class PrestaShop implements JsonSerializable
     private ?string $xmlDownloadUrl = null;
     private ?string $zipMD5 = null;
 
-    public function __construct(string $version)
+    public function __construct(string $version, string $flavor = self::FLAVOR_OPEN_SOURCE)
     {
+        if ($flavor !== self::FLAVOR_CLASSIC && $flavor !== self::FLAVOR_OPEN_SOURCE) {
+            throw new InvalidArgumentException(sprintf('Invalid flavor "%s" provided. Accepted values are: "%s", "%s".', $flavor, self::FLAVOR_CLASSIC, self::FLAVOR_OPEN_SOURCE));
+        }
```

Comment:
In the case we have more possible values in the future, and to make sure the test is always aligned with the error message, what do you think about having a list to reuse?

```suggestion
        if (!in_array($flavor, self::FLAVOR_POSSIBLE_VALUES) {
            throw new InvalidArgumentException(sprintf('Invalid flavor "%s" provided. Accepted values are: "%s".', $flavor, implode(', ', self::FLAVOR_POSSIBLE_VALUES)));
        }
```

and at the top of the class

```php
public const FLAVOR_POSSIBLE_VALUES = [self::FLAVOR_OPEN_SOURCE, self::FLAVOR_CLASSIC];
```